src/codegeneration/PlaceholderParser.js

```diff
@@ -260,7 +260,7 @@ export class PlaceholderTransformer extends ParseTreeTransformer {
           this.transformIdentifierExpression(tree.expression);
       if (transformedExpression === tree.expression)
         return tree;
-      if (transformedExpression.isStatement())
+      if (transformedExpression.isStatementListItem())
         return transformedExpression;
```

Comment:
FunctionDeclaration is not a Statement but a Declaration and StatementListItem is the union of those two.